docs/api_specs/includes/watcher_api_specs.md

```diff
@@ -0,0 +1,691 @@
+competing_input_index | integer | Index of the double-spent input in the competing transaction
+
+#### `invalid_ife_challenge`
+> A invalid_ife_challenge event
+
+```json
+{
+    "event": "invalid_ife_challenge",
+    "details": {
+        "inflight_blocknum"  : 10000,
+        "inflight_tx_index"  : 12,
+    }
+}
+```
+
+A canonical in-flight exit has been challenged. The challenge should be responded to. 
+<aside class="warning"> Not Implemented Yet.</aside> 
+
+Event details:
+Attribute | Type | Description
+--------- | ------- | -----------
+inflight_blocknum | integer | Block number of the in-flight transaction
+inflight_tx_index | integer | Index of the in-flight transaction
+
+#### `piggyback_available`
+> A piggyback_available event
+
+```json
+{
+    "event": "piggyback_available",
+    "details": {
+        "inflight_blocknum"  : 10000,
+        "inflight_tx_index"  : 12,
+        "inflight_io_index"  : 1
+    }
+}
+```
+
+An in-flight exit has been started and can be piggybacked 
+<aside class="warning"> Not Implemented Yet.</aside> 
+
+Event details:
+Attribute | Type | Description
+--------- | ------- | -----------
+inflight_blocknum | integer | Block number of the in-flight transaction
+inflight_tx_index | integer | Index of the in-flight transaction
+inflight_io_index | integer | Index of the input or output that can be exitted.
+
+#### `invalid_piggyback`
+> A invalid_piggyback event
+
+```json
+{
+    "event": "invalid_piggyback",
+    "details": {
+        "type": "output",
+        "inflight_blocknum"  : 10000,
+        "inflight_tx_index"  : 12,
+        "inflight_io_index"  : 1,
+        "spending_blocknum"  : 10000,
+        "spending_tx_index"  : 10,
+        "spending_input_index"  : 0,
+    }
+}
+```
+
+An invalid piggyback is in process. Should be challenged.
+<aside class="warning"> Not Implemented Yet.</aside>
+
+Event details:
+Attribute | Type | Description
+--------- | ------- | -----------
+type | string | Indicates whether the invalid piggyback is on an "input" or "output"
+inflight_blocknum | integer | Block number of the in-flight transaction
+inflight_tx_index | integer | Index of the in-flight transaction
+inflight_io_index | integer | Index of the input or output of the in-flight transaction
+spending_blocknum | integer | Block number of the spending transaction
+spending_tx_index | integer | Index of the spending transaction
+spending_input_index | integer | Index of the spent input iofn the spending transaction
+
+
+## Inflight Exit - Get Exit Data
+
+```shell
+curl http://localhost:4000/inflight_exit.get_data -d '{"txhash": "bdf562c24ace032176e27621073df58ce1c6f65de3b5932343b70ba03c72132d"}'
+```
+
+```elixir
+// TODO
+```
+
+```javascript
+// TODO
+```
+
+> The above command returns JSON document:
+
+```json
+{
+    "version": "1",
+    "success": true,
+    "data": {
+        "txbytes": "F847010180808080940000...",
+        "sigs": "7C29FB8327F60BBFC62...",
+        "input_txs" : [
```

Comment:
this is more elegant, but the convention is to abide by whatever the contract requires (we can also consider dropping that convention), which would be two equal-sized lists of `input_txs` and `input_proofs`